fgrpc/grpcrunner.go

```diff
@@ -279,6 +295,11 @@ func grpcDestination(dest string) (parsedDest string) {
 		port = fnet.StandardHTTPSPort
 		log.Infof("stripping https scheme. grpc destination: %v. grpc port: %s",
 			parsedDest, port)
+	case strings.HasPrefix(dest, fnet.PrefixXDS):
+		if _, _, err := net.SplitHostPort(dest[len(fnet.PrefixXDS):]); err == nil {
+			return dest
+		}
+		return dest + fnet.NormalizePort(fnet.DefaultGRPCPort)
```

Comment:
that's not quite right for the case there would be a port in the xds://url ?
why not letting it fall through like the rest of the code after remove `xds://` like `http://` ?

Reply:
This does: If there is a port, just keep it; but for no port use the default port. I have a test to check that it works.

I think the reason I didn't fallthrough is it would strip the `xds:///` which indicates to gRPC to use the resolver.

Reply:
can you give a full example of a couple of url; or point me to some; I'm very unfamiliar with xds

Reply:
reason I ask is
```
	if _, _, err := net.SplitHostPort(parsedDest); err == nil {
```
below is supposed to handle that

Reply:
Updated the test. AFAICT this is just a bit of code-duplication here to avoid adding something like `if xds { preserve scheme = true } ... if preserve scheme { ... }` in the outer block.